src/mw-inputs/directives/mw_autocomplete.js

```diff
@@ -0,0 +1,323 @@
+ * @param {mwRequired} boolean whether input is required
+ * @param {mwDisabled} boolean whether input is disabled
+ *
+ * @example ```html
+ * <!-- Lets the user browse through all availble users and set a user model -->
+ * <div mw-autocomplete
+ *      mw-model="enrollment.attributes.user"
+ *      mw-model-label-attr="name"
+ *      mw-options-collection="users"
+ *      mw-search-property="search"
+ *      mw-options-label-key="name"
+ *      mw-required="true"
+ *      mw-placeholder="Select a user">
+ *  </div>
+ * ```
+ */
+  .directive('mwAutocomplete', function ($q, $timeout, i18n) {
+    return {
+      scope: {
+        mwModel: '=',
+        mwModelAttr: '@',
+        mwModelLabelKey: '@',
+        mwSearchProperty: '@',
+        mwOptionsCollection: '=',
+        mwOptionsKey: '@',
+        mwOptionsLabelKey: '@',
+        mwOptionsLabelI18nPrefix: '@?',
+        mwMinSearchLength: '=?',
+        mwPlaceholder: '@?',
+        mwRequired: '=?',
+        mwDisabled: '=?'
+      },
+      replace: true,
+      templateUrl: 'uikit/mw-inputs/directives/templates/mw_autocomplete.html',
+      link: function (scope, el) {
+        var orgInputPadding = el.find('input').css('padding-left');
+
+        scope.viewModel = {
+          searchVal: '',
+          isSearching: false,
+          searchActive: false
+        };
+
+        var scrollToSelected = function (toBottom) {
+          var selected = scope.mwOptionsCollection.selectable.getSelected().first();
+          if (selected) {
+            var selectedEl = el.find('#' + selected.cid),
+              scrollContainerEl = el.find('.auto-complete-holder');
+
+            if (selectedEl.length) {
+              var selectedElOffset = selectedEl.offset().top,
+                selectedElHeight = selectedEl.outerHeight(true),
+                scrollContainerElOffset = scrollContainerEl.offset().top,
+                scrollContainerHeight = scrollContainerEl.height(),
+                scrollContainerScroll = scrollContainerEl.scrollTop();
+
+              var selY = selectedEl[0].getBoundingClientRect().y,
+                scrollConY = scrollContainerEl[0].getBoundingClientRect().y;
+
+              if (selY + selectedElHeight <= scrollConY || selY >= scrollConY + scrollContainerHeight) {
+                if (toBottom) {
+                  scrollContainerEl.scrollTop(((selectedElOffset - scrollContainerElOffset) - scrollContainerHeight + scrollContainerScroll) + selectedElHeight);
+                } else {
+                  scrollContainerEl.scrollTop(((selectedElOffset - scrollContainerElOffset) + scrollContainerScroll));
+                }
+              }
+            }
+          }
+        };
+
+        var setInputPadding = function () {
+          $timeout(function () {
+            if (scope.getSelectedModelLabel()) {
+              el.find('input').css('padding-left', el.find('.selected-item').outerWidth(true) + 15);
```

Comment:
`.find()` and `.css()` are also methods provided by [angulars jquery lite](https://docs.angularjs.org/api/ng/function/angular.element).  And in this case of dynamic style calculation its easier than ng-style. ng-class would not work because of the dynamic width

Reply:
I dont get your point with ` .find() and .css() are also methods...` yes, they are, i know. And that fact doesn't make it better.

I ask you again: Is there a better approach than using heavily jQuery. And one approach could be ng-style and ng-class. 

I mean from the performance view you always run through the DOM searching your elements to apply a style. The question should be: Is it necessary, or is there a way to apply the style without the search